Content.Goobstation.Shared/Xenobiology/Systems/XenoVacuumSystem.cs

```diff
@@ -0,0 +1,214 @@
+using Content.Shared.Hands;
+using Content.Shared.Humanoid;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory;
+using Content.Shared.Jittering;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Nutrition.EntitySystems;
+using Content.Shared.Popups;
+using Content.Shared.Stunnable;
+using Content.Shared.Throwing;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Configuration;
+using Robust.Shared.Containers;
+using Robust.Shared.Network;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Goobstation.Shared.Xenobiology.Systems;
+
+/// <summary>
+/// This handles the XenoVacuum and it's interactions.
+/// </summary>
+public sealed partial class XenoVacuumSystem : EntitySystem
+{
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+    [Dependency] private readonly EmagSystem _emag = default!;
+    [Dependency] private readonly ThrowingSystem _throw = default!;
+    [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    private ISawmill _sawmill = default!;
+
+    private EntityQuery<HumanoidAppearanceComponent> _humanoidQuery;
+    private EntityQuery<MobStateComponent> _mobQuery;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<XenoVacuumTankComponent, ComponentInit>(OnTankInit);
+        SubscribeLocalEvent<XenoVacuumTankComponent, ExaminedEvent>(OnTankExamined);
+
+        SubscribeLocalEvent<XenoVacuumComponent, GotEmaggedEvent>(OnVacEmagged);
+
+        SubscribeLocalEvent<XenoVacuumComponent, GotEquippedHandEvent>(OnEquippedHand);
+        SubscribeLocalEvent<XenoVacuumComponent, GotUnequippedHandEvent>(OnUnequippedHand);
+
+        SubscribeLocalEvent<XenoVacuumComponent, AfterInteractEvent>(OnXenoVacuum);
+
+        _sawmill = Logger.GetSawmill("Xenobiology");
+
+        _humanoidQuery = GetEntityQuery<HumanoidAppearanceComponent>();
+        _mobQuery = GetEntityQuery<MobStateComponent>();
+    }
+
+    private void OnTankInit(Entity<XenoVacuumTankComponent> tank, ref ComponentInit args) =>
+        tank.Comp.StorageTank = _containerSystem.EnsureContainer<Container>(tank, "StorageTank");
+
+    private void OnTankExamined(Entity<XenoVacuumTankComponent> tank, ref ExaminedEvent args)
+    {
+        if (!args.IsInDetailsRange)
+            return;
+
+        foreach (var ent in tank.Comp.StorageTank.ContainedEntities)
+        {
+            var text = Robust.Shared.Localization.Loc.GetString("xeno-vacuum-examined", ("ent", ent));
```

Comment:
HOLY GOIDACODO :sob:
```suggestion
            var text = Loc.GetString("xeno-vacuum-examined", ("ent", ent));
```